src/test/java/uk/gov/register/store/postgres/PostgresDriverTransactionalTest.java

```diff
@@ -1,89 +0,0 @@
-package uk.gov.register.store.postgres;
-
-import org.junit.Test;
-
-import java.util.Optional;
-import java.util.function.Consumer;
-
-import static java.util.Collections.emptyList;
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.empty;
-import static org.hamcrest.core.Is.is;
-import static org.mockito.Mockito.when;
-
-public class PostgresDriverTransactionalTest extends PostgresDriverTestBase {
-
-    @Test
-    public void insertRecordShouldNotCommitData() {
-        PostgresDriverTransactional postgresDriver = new PostgresDriverTransactional(
-                handle, h -> recordQueryDAO, h -> currentKeysUpdateDAO);
-
-        postgresDriver.insertRecord("DE", 1);
-
-        assertThat(currentKeys, is(empty()));
-    }
-
-    @Test
-    public void getRecordShouldAlwaysCommitStagedData() {
```

Comment:
I have written tests that assert the correct output is produced but are deliberately agnostic to whether the data is flushed to the database or not.

As long as the overall transaction succeeds (tested in PostgresRegisterTransactionalFunctionalTest) that should be all we need.